### PR TITLE
lexers: devicetree: allow node label without root node

### LINKED ISSUE #1754: Devicetree lexer does not node label without root node
It looks like the `devicetree` lexer throws some syntax warnings when trying to render code snippets like:

```devicetree
foo: foo@1234 {
    a = <1000 2000 3000>; /* array */
    b = [aa bb cc dd];    /* uint8-array */
    c = "bar", "baz";     /* string-array */
};
```

That is, something like `foo: bar@xxx {};`. However, it works if enclosed on a root node `\{};`:

```
/{
    foo: foo@1234 {
        a = <1000 2000 3000>; /* array */
        b = [aa bb cc dd];    /* uint8-array */
        c = "bar", "baz";     /* string-array */
    };
};
```

Ref. https://github.com/zephyrproject-rtos/zephyr/pull/33417

### PR DESCRIPTION
**Pull request by gmarull**
It is sometimes useful to render DeviceTree snippets such as:

```
foo: bar@1234 {
    foo = "bar";
};
```

However, the snippet shown above does not render properly unless it is
enclosed on a root node, i.e.

```
/{
    foo: bar@1234 {
        foo = "bar";
    };
};
```

Fixes #1754

### CODE REVIEW

**Comment by birkenfeld** [comment]
Thanks for the PR! Can you add a test snippet to exercise the new behavior?

**Comment by gmarull** [comment]
@birkenfeld done

File: tests/test_devicetree_lexer.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+@pytest.fixture(scope='module')
+def lexer():
+    yield DevicetreeLexer()
+
+
+def test_fragment_out_of_root_node(lexer):
+    """Validate that a devicetree fragment out of a root node is parsed correctly."""
+
+    fragment = 'nodelabel: node@0 { foo = "bar"; };'
```
Comment (mbolivar-nordic) [inline tests/test_devicetree_lexer.py:23]: Just confirming, but these fragments all work as expected (i.e. are valid)  with this PR, right?

1. `nodelabel: node { foo = "bar"; };`
2. `nodelabel1: nodelabel2: node@0 { foo = "bar"; };`
3. `nodelabel1: nodelabel2: node { foo = "bar"; };`

Comment (gmarull) [reply tests/test_devicetree_lexer.py:23]: good spot, multi label is not parsed correctly.

Comment (gmarull) [reply tests/test_devicetree_lexer.py:23]: actually, `nodelabel1: nodelabel2: node { foo = "bar"; };` is parsed correctly even with current implementation

Comment (gmarull) [reply tests/test_devicetree_lexer.py:23]: current lexer doesn't seem to play well when we have the combination of `label + node + @ + addr` out of root node.

Comment (mbolivar-nordic) [reply tests/test_devicetree_lexer.py:23]: Makes sense since that's only valid in a snippet.

Comment (gmarull) [reply tests/test_devicetree_lexer.py]: latest changes should fix all the cases mentioned above.

**Pull request merged**